src/Word.jl

```diff
@@ -0,0 +1,68 @@
+abstract type Word{T} <: AbstractVector{T} end
+
+
+
+#interface for morphism T->Int
+function to_int{T}(t::T) end
+#interface for morphism Int->T
+function from_int{T}(t::T) end
+#interface for ordering on T
+function <=(a::T,b::T) end
+#empty string as monoid neutral element
+Base.one(::Type{Word{T}}) where {T}  = Word(Vector{T}())
+#overload for <= that implements word ordering
+function <=(a::Word{T},b::Word{T})
+#overload for * as concatenation
+function *(a::Word{T},b::Word{T}) where {T} Word(a.ptrs+b.ptrs) end
```

Comment:
here we need something more:
```julia
Base.:*(a::Word{T}, b::Word{T}) where T = Word([a.ptrs; b.ptrs]) # syntax for appending
```
the syntax for overloading `*` (it needs to be prepended with `:`) is a little cumbersome